src/MediatR/Mediator.cs

```diff
@@ -73,20 +88,65 @@ public void Publish(INotification notification)
 
         public Task PublishAsync(IAsyncNotification notification)
         {

```

Comment:
The problem with this is that I'd then have to update every call site in my code to get the behaviour I want - since I want sequential behaviour whenever events are published.

If the default option on the Mediator is configured correctly then there would be no breaking change in behaviour irrespective of how the method is called.